src/server/svr_jobfunc.c

```diff
@@ -6114,9 +6214,12 @@ recreate_exec_vnode(job *pjob, char *vnodelist, char *err_msg,
 		(void)update_resources_list(pjob, ATTR_l,
 			JOB_ATR_resource, new_exec_vnode, INCR, 0,
 				JOB_ATR_resource_orig);
+	} else {
+		log_err(-1, __func__, "new_exec_vnode is null or empty string");
+		goto recreate_exec_vnode_exit;
 	}
 
-	if (new_deallocated_execvnode[0] != '\0') {
+	if ((keep_select == NULL) && new_deallocated_execvnode && (new_deallocated_execvnode[0] != '\0')) {
```

Comment:
Please be consistent with how you perform these checks.

Reply:
Fixed in commit https://github.com/PBSPro/pbspro/pull/1452/commits/8f309272ad00f7587c830cad07238cb40598e24a